tests/optional_kernel_features/kernel_features_separate_unit.cpp

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides function in separate unit for tests of the optional kernel features
+//  with SYCL_EXTERNAL function
+//
+*******************************************************************************/
+
+#include "../common/common.h"
+
+namespace kernel_features_common {
+template <typename T, sycl::aspect aspect>
+[[sycl::device_has(aspect)]] SYCL_EXTERNAL void
+use_feature_function_external_decorated();
+
+template <>
+SYCL_EXTERNAL void
+use_feature_function_external_decorated<sycl::half, sycl::aspect::fp16>() {
+  float temp = 42.25f;
+  sycl::half feature(temp);
+  feature += 42;
+}
+
+template <>
+SYCL_EXTERNAL void
+use_feature_function_external_decorated<double, sycl::aspect::fp64>() {
+  float temp = 42.25f;
+  double feature(temp);
+  feature += 42;
+}
+
+using AtomicRefT =
+    sycl::atomic_ref<unsigned long long, sycl::memory_order::relaxed,
+                     sycl::memory_scope::device>;
+template <>
+SYCL_EXTERNAL void
```

Comment:
Should an implementation define this macro even if it doesn't support the functionality? If not (and the spec suggests so https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_sycl_linker), this test will not pass on that implementation, so it should be made optional somehow. Related to the discussion in https://github.com/KhronosGroup/SYCL-CTS/pull/285#pullrequestreview-885634602 .

Reply:
@andreyromanof, can we remove optional_kernel_features test from all the filters?

Reply:
I think right now we cant remove these tests, due to issues that I mentioned in the description of the PR.